main/src/ca/uwaterloo/flix/api/Bootstrap.scala

```diff
@@ -424,6 +425,55 @@ class Bootstrap(val projectPath: Path, apiKey: Option[String]) {
     }
   }
 
+  /**
+    * Sequentially deletes all `.class` files in the build directory if ALL files (recursively) in the build directory
+    * are `.class` files.
+    * Aborts if any other file was found.
+    */
+  def clean(): Result[Unit, BootstrapError] = {
+    val buildDir = getBuildDirectory(projectPath)
+    val root = Path.of("/").normalize()
+
+    if (buildDir == root) {
+      return Err(BootstrapError.FileError("unexpected build dir"))
+    }
+
+    val files = FileOps.getFilesIn(buildDir, Int.MaxValue)
+    for (file <- files) {
+      if (!FileOps.checkExt(file, "class")) {
```

Comment:
two questions:

- What about html doc? and css? (You can run the doc command)
- What about the artifact directory?

Reply:
Let's focus on class files first and see where that gets us then we figure out other extensions after